packages/vscode-docker-registries/src/clients/GenericRegistryV2/GenericRegistryV2DataProvider.ts

```diff
@@ -57,19 +72,58 @@ export class GenericRegistryV2DataProvider extends RegistryV2DataProvider {
         return this.authenticationProviders.get(registry)!;
     }
 
-    private addTrackedRegistry(): void {
-        //const trackedRegistryStrings = this.extensionContext.globalState.get<string[]>(TrackedRegistriesKey, []);
+    public async addTrackedRegistry(): Promise<void> {
+        const wizardContext: GenericRegistryV2WizardContext = {
+            registryPrompt: vscode.l10n.t('Registry URL'), // TODO: change prompt
+            usernamePrompt: vscode.l10n.t('Registry Username'), // TODO: change prompt
+            secretPrompt: vscode.l10n.t('Registry Password or Personal Access Token'), // TODO: change prompt
+        };
+
+        const wizard = new RegistryWizard(
+            wizardContext,
+            [
+                new GenericRegistryV2WizardPromptStep(),
+                new RegistryWizardUsernamePromptStep(),
+                new RegistryWizardSecretPromptStep(),
+            ],
+            new vscode.CancellationTokenSource().token
+        );
+
+        await wizard.prompt();
+
+        if (!wizardContext.registryUri) {
+            throw new Error('Registry URL is invalid');
+        }
+
+        const registryUriString = wizardContext.registryUri.toString();
+
+        // store registry url in memento
+        const trackedRegistryStrings = this.extensionContext.globalState.get<string[]>(TrackedRegistriesKey, []);
+        const index = trackedRegistryStrings.findIndex(r => r === registryUriString);
+        if (index > -1) { // if found, throw error
+            throw new Error('Registry already exists'); //TODO: localize and see if there is a better way to handle this

```

Comment:
Doesn't have to be in this PR, but it would be ideal for re-adding the "same" registry to just overwrite

Reply:
i.e. make adding idempotent